starknet_devnet/blueprints/base.py

```diff
@@ -195,6 +195,14 @@ async def create_block():
     return Response(block.dumps(), status=200, mimetype="application/json")
 
 
+@base.route("/create_block_on_demand", methods=["POST"])
+async def create_block_on_demand():
+    """Create block on demand"""
+    block = await state.starknet_wrapper.store_mempool_transactions()
```

Comment:
What happens if no txs in the mempool?

Reply:
An empty block will be generated.

Reply:
Ok, this should be documented then

Reply:
fixed in dd8dc8d8b07198cb21e48665772b79891aa101d9

Reply:
Do users know what a mempool is? Also the genesis block info looks out of place: it looks like it's related to the previous sentence, but I don't think it should be. Does it even have to be mentioned here?

Reply:
Mempool is a common term in blockchain terminology, they can google that also.

I think that it's worth mentioning so the user, doesn't need to execute `/create_block_on_demand` immediately after devnet starts.

Reply:
> Mempool is a common term in blockchain terminology, they can google that also.

Ok, makes sense.

> I think that it's worth mentioning so the user, doesn't need to execute `/create_block_on_demand` immediately after devnet starts.

Sure that piece of information should be in the docs, but should it be in the very paragraph where it is right now. When I see this:
> In case of no transactions in mempool new empty block will be generated. The genesis block will be generated normally.

I consider the genesis block generation to be related to the mempool being empty. We are not trying to say that, right?

Reply:
Different blockchains generate genesis blocks differently, sometimes genesis block contains transactions or some state at the beginning. I just wanted to point out that it this case with this flag genesis block will be generated and will be empty.

Reply:
ok, resolving this conversation